app/scripts/lib/start-up-errors/start-up-errors.ts

```diff
@@ -0,0 +1,35 @@
+import log from 'loglevel';
+
+/**
+ * Attempts to post a `message` to the given `port` via `port.postMessage`. If
+ * the postMessage call fails due to an exception it will ignore it by catching
+ * and then logging it.
+ *
+ * @param port - The port to post the message to.
+ * @param method - The method to call on the port.
+ * @param params - The parameters to include in the message.
+ * @returns Returns true if the message was posted successfully, false otherwise.
+ */
+export function tryPostMessage(
```

Comment:
This function has been moved here from `app/scripts/lib/state-corruption/state-corruption-recovery.ts`. Functionality is unchanged.